### DISCUSSION TITLE
SSO with azure AD from RC101 to 9.2.0 doesn't work anymore

### DISCUSSION THREAD

**Discussion by KZ-Millennium** [directus/directus#11316] [Archive]
Hello, I updated directus from RC101 to 9.2.0 and it seems the sso with azure AD has some breaking changes.

I didn't find any information in release note.

Here is the original config (which worked in RC101)

```
AUTH_MICROSOFT_CLIENT_ID=CLIENTID
AUTH_MICROSOFT_CLIENT_SECRET=SECRET
AUTH_MICROSOFT_AUTHORIZE_URL=https://login.microsoftonline.com/TENANTID/oauth2/v2.0/authorize
AUTH_MICROSOFT_SCOPE=openid email
AUTH_PROVIDERS=microsoft
AUTH_MICROSOFT_DRIVER=oauth2
AUTH_MICROSOFT_ACCESS_URL=https://login.microsoftonline.com/TENANTID/oauth2/v2.0/token
```

I have tried adding some new fields but I keep getting "INVALID_USER"

```
AUTH_MICROSOFT_ALLOW_PUBLIC_REGISTRATION=false
AUTH_MICROSOFT_PROFILE_URL=https://graph.microsoft.com/oidc/userinfo
AUTH_DISABLE_DEFAULT=false
AUTH_MICROSOFT_CLIENT_ID=CLIENTID
AUTH_MICROSOFT_CLIENT_SECRET=SECRET
AUTH_MICROSOFT_AUTHORIZE_URL=https://login.microsoftonline.com/TENANTID/oauth2/v2.0/authorize
AUTH_MICROSOFT_ICON=microsoft
AUTH_MICROSOFT_EMAIL_KEY=email
AUTH_MICROSOFT_SCOPE=openid profile email
AUTH_PROVIDERS=microsoft
AUTH_MICROSOFT_DRIVER=oauth2
AUTH_MICROSOFT_ACCESS_URL=https://login.microsoftonline.com/TENANTID/oauth2/v2.0/token
```
Here is the config of one of the user in 9.2.0

![image](https://user-images.githubusercontent.com/74610514/151525572-40a299f9-a59b-4aa5-8a88-bf4155f0fa68.png)

Can you help me figuring out what I'm doing wrong ?

Thank you !

**Comment by benhaynes** [comment]
CC @aidenfoxx

**Reply by aidenfoxx** [reply]
This was caused by a sneaky change in the `openid-client` library which prevents using an OpenID provider in an OAuth2 configuration. It's good in the sense that it forces users into the more secure OpenID flow, but difficult to communicate well. The most minimal solution is that users don't specify "openid" in their scope when using the "oauth2" driver. The ideal solution is they just use the correct "openid" driver.

**Comment by aidenfoxx** [comment]
This is caused by a change in `openid-client` (https://github.com/panva/node-openid-client/commit/92ffee5c63dc31fb578c731572cf0f83a7b53f1d). It is no longer possible to use an OpenID provider in OAuth2 mode. There was a discussion about this here https://github.com/directus/directus/discussions/11048

You simply need to update your config to use OpenID instead of OAuth2 (example here https://github.com/directus/directus/discussions/10511) or remove openid from your scope.

**Reply by KZ-Millennium** [reply]
Works perfectly ! Thank you for the answer !

**Comment by biapar** [comment]
Could you post your config?

**Reply by KZ-Millennium** [reply]
sure:
```
AUTH_MICROSOFT_CLIENT_ID=CLIENT_ID
AUTH_MICROSOFT_CLIENT_SECRET=CLIENT_SECRET
AUTH_MICROSOFT_DRIVER=openid
AUTH_MICROSOFT_EMAIL_KEY=email
AUTH_MICROSOFT_IDENTIFIER_KEY=email
AUTH_MICROSOFT_ISSUER_URL=https://login.microsoftonline.com/TENANT_ID/v2.0/.well-known/openid-configuration
AUTH_MICROSOFT_SCOPE=openid email
AUTH_PROVIDERS=microsoft
```
